garak/probes/visual_jailbreak.py

```diff
@@ -33,55 +37,112 @@ class FigStep(Probe):
             raise ValueError(
                 f"Incorrect generator input modality {generator.modality['in']}, expect {self.modality['in']} for this probe."
             )
-        self.prompts = [{'text': prompt['text'], 'image': str(_config.transient.basedir / prompt['image'])} for prompt in self.prompts]
-        if len(self.prompts) != 500:
-            logging.warning(f"Incorrect number ({len(self.prompts)}) of image prompts of SafeBench dataset, expect 500.")
+        self.prompts = [
+            {
+                "text": prompt["text"],
+                "image": str(_config.transient.basedir / prompt["image"]),
+            }
+            for prompt in self.prompts
+        ]
+        if len(self.prompts) != len(self.safebench_image_filenames):
+            logging.warning(
+                f"Incorrect number ({len(self.prompts)}) of image prompts of SafeBench dataset, expect {len(self.safebench_image_filenames)}."
+            )
+
         return super().probe(generator)
 
-class FigStep80(FigStep, Probe):
+
+class FigStepTiny(FigStep, Probe):
     active = True
-    __doc__ = FigStep.__doc__ + " - limited to 80 examples"
+
+    __doc__ = FigStep.__doc__ + " - Tiny version"
+
+    safebench_image_catalog = (
+        _config.transient.basedir / "resources" / "safebenchtiny_filenames.txt"
+    )
 
     def probe(self, generator):
         if not isinstance(generator, Generator):
-            raise ValueError(
-                "Incorrect class type of incoming argument `generator`."
-            )
-        if not generator.modality['in'] == self.modality['in']:
+            raise ValueError("Incorrect class type of incoming argument `generator`.")
+        if not generator.modality["in"] == self.modality["in"]:
             raise ValueError(
                 f"Incorrect generator input modality {generator.modality['in']}, expect {self.modality['in']} for this probe."
             )
-        self.prompts = [{'text': prompt['text'], 'image': str(_config.transient.basedir / prompt['image'])} for prompt in self.prompts]
+        self.prompts = [
+            {
+                "text": prompt["text"],
+                "image": str(_config.transient.basedir / prompt["image"]),
+            }
+            for prompt in self.prompts
+        ]
 
         # We want 8 image prompts for 10 topics seperately
         # Assume we have downloaded full SafeBench dataset with 500 image prompts
         # https://github.com/ThuCCSLab/FigStep#:~:text=500%20questions%20on%2010%20topics
-        self.prompts = [prompt for prompt in self.prompts if 1 <= int(os.path.basename(prompt['image']).split('_')[3]) <= 8]
-        if len(self.prompts) != 80:
-            logging.warning(f"Incorrect number ({len(self.prompts)}) of image prompts of SafeBench dataset, expect 80.")
+        self.prompts = [
+            prompt
+            for prompt in self.prompts
+            if 1 <= int(os.path.basename(prompt["image"]).split("_")[3]) <= 8
```

Comment:
Origin version retain first 8 image prompts of each 10 types, 80 image prompts would be there. It can be change in https://github.com/leondz/garak/pull/587/commits/16a59724d062e60e5db70d9de97d84f4eaf24c7d that we retain tiny version of SafeBench.